File: lib/ui/student/components/summary_card.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

import '../../../domain/model/student_summary.dart';
import '../../theme/app_colors.dart';

class SummaryCard extends StatelessWidget {
  const SummaryCard({
    super.key,
    required this.summary,
    required this.onDownload,
  });

  final StudentSummary summary;
  final VoidCallback onDownload;

  @override
  Widget build(BuildContext context) {
    final text = Theme.of(context).textTheme;
    final title = summary.section == 'All'
        ? summary.batch
        : '${summary.batch}  ${summary.section}';
    return DecoratedBox(
      decoration: BoxDecoration(
        color: peach,
        borderRadius: BorderRadius.circular(32),
      ),
      child: Padding(
        padding: const EdgeInsets.all(22),
        child: Row(
          children: [
            Expanded(
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(title, style: text.headlineSmall),
                  const SizedBox(height: 6),
                  Text(
                    '${summary.totalCourses} courses  ·  ${summary.classesPerWeek} classes',
                    style: text.bodyMedium,
                  ),
                ],
              ),
            ),
            Material(
              color: ink,
              shape: const CircleBorder(),
              child: InkWell(
                customBorder: const CircleBorder(),
                onTap: () {
                  HapticFeedback.mediumImpact();
                  onDownload();
                },
                child: const SizedBox(
                  width: 52,
                  height: 52,
                  child: Icon(Icons.download_outlined, color: onInk, size: 24),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: app/src/main/java/com/ababilx/routinescrapper/ui/student/components/SummaryCard.kt
package com.ababilx.routinescrapper.ui.student.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.ababilx.routinescrapper.domain.model.StudentSummary
import com.ababilx.routinescrapper.ui.icons.AppIcons
import com.ababilx.routinescrapper.ui.theme.Ink
import com.ababilx.routinescrapper.ui.theme.OnInk
import com.ababilx.routinescrapper.ui.theme.Peach
import com.ababilx.routinescrapper.ui.theme.RdiuTypography
import com.ababilx.routinescrapper.ui.theme.TextMuted

@Composable
fun SummaryCard(
    summary: StudentSummary,
    onDownload: () -> Unit,
    modifier: Modifier = Modifier,
) {
    val title = if (summary.section == "All") summary.batch else "${summary.batch}  ${summary.section}"
    Surface(
        modifier = modifier.fillMaxWidth(),
        color = Peach,
        shape = RoundedCornerShape(32.dp),
    ) {
        Row(
            modifier = Modifier.padding(22.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.SpaceBetween,
        ) {
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(6.dp),
            ) {
                Text(title, style = RdiuTypography.headlineSmall)
                Text(
                    "${summary.totalCourses} courses  ·  ${summary.classesPerWeek} classes",
                    style = RdiuTypography.bodyMedium,
                    color = TextMuted,
                )
            }
            Surface(
                modifier = Modifier
                    .size(52.dp)
                    .clickable(onClick = onDownload),
                color = Ink,
                shape = CircleShape,
            ) {
                Icon(
                    imageVector = AppIcons.Download,
                    contentDescription = "Download PDF",
                    tint = OnInk,
                    modifier = Modifier.padding(14.dp),
                )
            }
        }
    }
}
